src/mpid/ch4/netmod/ucx/ucx_types.h

```diff
@@ -18,8 +18,9 @@
 #define UCP_PEER_NAME_MAX         HOST_NAME_MAX
 
 /* Active Message Stuff */
-#define MPIDI_UCX_MAX_AM_EAGER_SZ      (16*1024)
+#define MPIDI_UCX_MAX_AM_EAGER_SZ      (8000)   /* internal max RTS 8256 */
```

Comment:
what's the intention of this change?

Reply:
UCP internal RTS is limited to 8256 bytes -- no idea why the odd number. In our RMA code we decide whether to send in the flattened datatype as extended header based on the `AM_EAGER_SZ`. The extended header will be packed into RTS and blow up the UCP internal limit if the EAGER size is too big.

Reply:
Is it documented? I see that number as `SEG_SIZE` in the mm and ib transports. My best guess is it is 64 byte header + 8k copy buffer. I don't have a problem with the change. Maybe the comment could be updated to say this is a UCP RTS limit.